pkg/scheduler/factory/factory.go

```diff
@@ -232,7 +236,8 @@ type ConfigFactoryArgs struct {
 	StatefulSetInformer            appsinformers.StatefulSetInformer
 	ServiceInformer                coreinformers.ServiceInformer
 	PdbInformer                    policyinformers.PodDisruptionBudgetInformer
-	StorageClassInformer           storageinformers.StorageClassInformer
+	StorageClassInformer           storageinformersv1.StorageClassInformer
+	CSINodeInformer                storageinformersv1beta1.CSINodeInformer
```

Comment:
I am not 100% sure about the performance of scheduler with this change and since this feature is beta, this will be enabled by default. So, I feel less confident about this change going into 1.15.

Reply:
There were no scheduler benchmarks with volumes, I am preparing one at https://github.com/kubernetes/kubernetes/pull/77397.

Current numbers (5000 nodes each with CSINode instance, 1000 pods each with a CSI volume, with all alpha gates disabled, two measurements each):

With this PR:    30583423 and 37684721 ns/op
Without this PR: 33340480 and 36254432 ns/op

These numbers are jumping like crazy... will do proper stats on a stable system tomorrow.

Reply:
Finally I got some real numbers:

| Volume             | Original ns/op | With this PR ns/op | Increase |
| ------------------ | -------------- | ------------------ | -------- |
| In-tree volume     |  77432515      |  78835497          | 101.81%  |
| CSI volume         |  64654527      |  70410337          | 108.90%  |
| Migrated volume    |  80161751      |  78255729          | 97.62%   |

All numbers are form tests BenchmarkScheduling*PVs/5000Nodes/1000Pods-4 on 4-core GCE machine, 10 samples each (variation is low, +/- 2-3%)

CSI scheduling is 8% slower, the rest is IMO fine.

Reply:
I ran the tests with current version + enabled benchmark of a pod with Secrets volume (in two flavours, one with no CSINode instances and second with CSINode for each Node), results are not so nice:

Test | Before (ns/op) | After (ns/op) | Increase
-- | -- | -- | --
BenchmarkSchedulingInTreePVs | 79628055.7 | 79433974.9 | -0.24%
BenchmarkSchedulingCSIPVs | 65328092.6 | 70548380.8 | 7.99%
BenchmarkSchedulingMigratedInTreePVs | 80408231.1 | 80375741.1 | -0.04%
BenchmarkSchedulingSecrets | 30153507.9 | 33890242.5 | 12.39%
BenchmarkSchedulingSecretsWithCSINodes | 30181808.4 | 36221780.5 | 20.01%

Main culprits:

- feature gate checks
- VolumeLimits() call

We're working on optimizations.

Reply:
BenchmarkSchedulingSecretsWithCSINodes mistakenly turned on migration alpha gate while the others had it disabled.

I'm testing out some optimizations.